### PR TITLE
chore(CPlusPlus): add even odd algorithm

### PR DESCRIPTION
**Pull request by en1gm479**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
I have added the Algorithm to check for Even and Odd numbers in CplusPlus. I've added two methods for checking Even-Odd, one through Logical And operator (&) and other through Mod operator(%). 

This issue (#501) has been open for a long time and I have decided to close it by making this PR.

### CODE REVIEW

**Comment by en1gm479** [comment]
Spell corrected (sting->string) in algorithms/Java/Maths/roman-numerals.java which leads to failing of codepsell workflow.

**Comment by en1gm479** [comment]
@ming-tsai please review the PR

**Comment by en1gm479** [comment]
@mohitchakraverty @ming-tsai please review the PR.

File: algorithms/Java/Maths/roman-numerals.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * ----------------------------------------------------------------------------------------------------------------
  *
  * Constraints:
- *      The input sting should contain only the characters ('I', 'V', 'X', 'L', 'C', 'D', 'M').
+ *      The input string should contain only the characters ('I', 'V', 'X', 'L', 'C', 'D', 'M').
```
Comment (mohitchakraverty) [inline algorithms/Java/Maths/roman-numerals.java:13]: No need of making changes to this file. Its already fixed and merge

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Update your branch en1gm479:add-CheckEvenOdd with the recent changes done to MakeContributions:main
Then all the test cases will be passed.

**Review by mohitchakraverty — APPROVED** [review]

**Comment by mohitchakraverty** [comment]
Great work !

**Pull request merged**